test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ModelBuilder/SpecialCharactersLinkGenerationTests.cs

```diff
@@ -51,12 +51,12 @@ public IQueryable<SpecialCharactersLinkGenerationTestsModel> Get()
             return Todoes.AsQueryable();
         }
 
-        protected SpecialCharactersLinkGenerationTestsModel GetEntityByKey(string key)
```

Comment:
>GetEntityByKey [](start = 60, length = 14)

GetEntityByKey can't work now? It maybe a regression?

Reply:
The test had a bug and this method (`GetEntityByKey`) was not getting called, either because it was protected or because of the name, I don't remember. And the test case was not verifying the response of the request. So the test was getting an incorrect result but it was reporting a success. Specifically OData does not support single whitespace (`' '`) as key but the test marked it as supported. This was not related to my PR, it was a bug that existed in the test, but my PR helped expose the bug because of the stricter routing selection that I implemented that caused the request to return a 404 error instead of routing to the wrong method.